nominal/core/write_stream_base.py

```diff
@@ -71,6 +71,7 @@ def enqueue_from_dict(
         self,
         timestamp: str | datetime | IntegralNanosecondsUTC,
         channel_values: dict[str, float | str],
```

Comment:
I think this also needs to be a `Mapping`
```suggestion
        channel_values: Mapping[str, float | str],
```